src/component.js

```diff
@@ -205,21 +198,22 @@ export function enqueueRender(c) {
 		prevDebounce !== options.debounceRendering
 	) {
 		prevDebounce = options.debounceRendering;
-		(prevDebounce || defer)(process);
+		(prevDebounce || setTimeout)(process);
 	}
 }
 
 /** Flush the render queue by rerendering all queued components */
-function process() {
+async function process() {
 	let queue;
 	while ((process._rerenderCount = rerenderQueue.length)) {
 		queue = rerenderQueue.sort((a, b) => a._vnode._depth - b._vnode._depth);
 		rerenderQueue = [];
 		// Don't update `renderCount` yet. Keep its value non-zero to prevent unnecessary
 		// process() calls from getting scheduled while `queue` is still being consumed.
```

Comment:
Note that we are awaiting for each component render. Not waiting for them would result in "concurrent rendering", which causes problems like double rendering components.